pkg/detector/detector.go

```diff
@@ -899,12 +900,24 @@ func (d *ResourceDetector) GetMatching(resourceSelectors []policyv1alpha1.Resour
 
 // OnPropagationPolicyAdd handles object add event and push the object to queue.
 func (d *ResourceDetector) OnPropagationPolicyAdd(obj interface{}) {
-	d.policyReconcileWorker.Enqueue(obj)
+	policyObj := obj.(*policyv1alpha1.PropagationPolicy)
+	policyObj.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   policyv1alpha1.GroupVersion.Group,
+		Version: policyv1alpha1.GroupVersion.Version,
+		Kind:    policyv1alpha1.ResourceKindPropagationPolicy,
+	})
```

Comment:
I'm not sure but remember the controller-runtime fills the missing GVK.  Can you confirm it?

By the way, if it indeed needs to set the GVK, please add a comment to explain the reason.

Reply:
> I'm not sure but remember the controller-runtime fills the missing GVK. Can you confirm it?

It does not fill the GVK, I have also encountered it in my environment.

Reply:
> I'm not sure but remember the controller-runtime fills the missing GVK. Can you confirm it?

I used controller-runtime, kubernetes go-client, karmada go-client and kubernetes dynamic go-client to create informers for PropagationPolicy and Deployment to observe whether the objects triggered by the handler contain GVK. The results are as follows:
 | client type      | PropagationPolicy |Deployment|Unstructured|
| ----------- | ----------- |----------- |----------- |
| controller-runtime      | YES       |NO|-|
| kubernetes go-client   |-        |NO|-|
| karmada go-client  | YES        |NO|-|
|kubernetes dynamic go-client | YES|-|YES|